include/rocksdb/system_clock.h

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+
+#pragma once
+#include <rocksdb/rocksdb_namespace.h>
+#include <rocksdb/status.h>
+#include <stdint.h>
+
+#include <memory>
+
+#ifdef _WIN32
+// Windows API macro interference
+#undef GetCurrentTime
+#endif
+
+namespace ROCKSDB_NAMESPACE {
+struct ConfigOptions;
+
+// A SystemClock is an interface used by the rocksdb implementation to access
+// operating system time-related functionality.
+class SystemClock {
+ public:
+  virtual ~SystemClock() {}
+
+  static const char* Type() { return "SystemClock"; }
+
+  // Creates and configures a new SystemClock from the input options and id.
+  // MJR static Status CreateFromString(const ConfigOptions& config_options,
+  // MJR                               const std::string& id,
+  // MJR                               std::shared_ptr<SystemClock>* clock);
+
+  // The name of this system clock
+  virtual const char* Name() const = 0;
+
+  // Return a default SystemClock suitable for the current operating
+  // system.
+  static const std::shared_ptr<SystemClock>& Default();
+
+  // Returns the number of micro-seconds since some fixed point in time.
+  // It is often used as system time such as in GenericRateLimiter
+  // and other places so a port needs to return system time in order to work.
+  virtual uint64_t NowMicros() = 0;
+
+  // Returns the number of nano-seconds since some fixed point in time. Only
+  // useful for computing deltas of time in one run.
+  // Default implementation simply relies on NowMicros.
+  // In platform-specific implementations, NowNanos() should return time points
+  // that are MONOTONIC.
+  virtual uint64_t NowNanos() { return NowMicros() * 1000; }
+
+  // 0 indicates not supported.
```

Comment:
Appears to be "CPU time spent in current thread"

Reply:
I do not know that "NowMicros" implies CPU time for the current thread, does it?

Reply:
This is a comment on `NowCPUNanos` which is not documented :(

Reply:
Renamed NowCPUNanos to CPUNanos and added a comment.